src/components/ha-menu.ts

```diff
@@ -18,4 +38,8 @@ declare global {
   interface HTMLElementTagNameMap {
     "ha-menu": HaMenu;
   }
+
+  interface HTMLElementEventMap {
+    "close-menu": CloseMenuEvent;
```

Comment:
Why is this needed?

Reply:
Without this, I had the following lint error:

```
src/components/ha-menu.ts:15:10 - error TS2769: No overload matches this call.
  Overload 1 of 2, '(type: keyof HTMLElementEventMap, listener: (this: HTMLElement, ev: Event | HASSDomEvent<ShowDialogParams<unknown>> | HASSDomEvent<DialogClosedParams> | ... 43 more ... | HASSDomEvent<...>) => any, options?: boolean | ... 1 more ... | undefined): void', gave the following error.
    Argument of type '"close-menu"' is not assignable to parameter of type 'keyof HTMLElementEventMap'.
  Overload 2 of 2, '(type: string, listener: EventListenerOrEventListenerObject, options?: boolean | AddEventListenerOptions | undefined): void', gave the following error.
    Argument of type '(ev: CustomEvent<{ initiator: MenuItem; itemPath: MenuItem[]; reason: DefaultReasons; }>) => void' is not assignable to parameter of type 'EventListenerOrEventListenerObject'.
      Type '(ev: CustomEvent<{ initiator: MenuItem; itemPath: MenuItem[]; reason: DefaultReasons; }>) => void' is not assignable to type 'EventListener'.
        Types of parameters 'ev' and 'evt' are incompatible.
          Type 'Event' is missing the following properties from type 'CustomEvent<{ initiator: MenuItem; itemPath: MenuItem[]; reason: DefaultReasons; }>': detail, initCustomEvent

15     this.addEventListener("close-menu", this._handleCloseMenu);
            ~~~~~~~~~~~~~~~~
```

Reply:
Hmm, I figured material already put their events on the global map, but I guess not 🤷🏻‍♂️